storage/remote/otlptranslator/prometheusremotewrite/number_data_points.go

```diff
@@ -45,6 +45,7 @@ func (c *PrometheusConverter) addGaugeNumberDataPoints(ctx context.Context, data
 			nil,
 			true,
 			metadata,
+			0,
```

Comment:
[Nit] Could we use the corresponding constant for improved readability, and not rely on the constant's value?
```suggestion
			pmetric.AggregationTemporalityUnspecified,
```